mlir/tools/mlir-tblgen/AttrOrTypeFormatGen.cpp

```diff
@@ -816,14 +840,29 @@ static void guardOnAny(FmtContext &ctx, MethodBody &os, ParameterRange &&params,
   os.indent();
 }
 
+/// Generate code to guard printing on the presence of any optional format
+/// elements.
+template <typename FormatElemRange>
+static void guardOnAnyOptional(FmtContext &ctx, MethodBody &os,
+                               FormatElemRange &&args, bool inverted = false) {
+  guardOnAny(ctx, os,
+             llvm::make_filter_range(
+                 llvm::map_range(args,
+                                 [](FormatElement *el) {
+                                   return getStructParameterElement(el);
+                                 }),
```

Comment:
```suggestion
                                 getStructParameterElement),
```

Do you need to wrap getStructParameterElement, or can you just pass it in directly?